src/lib.rs

```diff
@@ -10,11 +10,34 @@ pub enum Error {
     Checksum,
 }
 
+/// Protected ID which is a 6 bit ID with two parity bits
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub struct PID(u8);
+
+impl PID {
+    /// Calculate the PID from an ID.
+    /// P0 = ID0 ⊕ ID1 ⊕ ID2 ⊕ ID4
+    /// P1 = ¬(ID1 ⊕ ID3 ⊕ ID4 ⊕ ID5)
+    pub fn from_id(id: u8) -> PID {
+        assert!(id < 64, "ID must be less than 64");
+        // count parity bits and check if they are even odd
+        let p0 = (id & 0b10111).count_ones() as u8 & 0b1;
+        let p1 = ((id & 0b111010).count_ones() as u8 + 1) & 0b1;
```

Comment:
I'm not sure if this is the most readable / straight forward implementation for the parity bits, but I wanted to avoid having to do `((id & 1 << 1) != 0) ^ ((id & 1 << 2) != 0) ...` and then convert from bool again to actually set the parity bits.